developerportal/templatetags/app_tags.py

```diff
@@ -57,3 +61,38 @@ def has_at_least_two_filters(filters):
     # least two of its keys' values are truthy, else False
     result = len([val for val in filters.values() if bool(val)]) >= 2
     return result
+
+
+@register.filter
+def filename_cachebreaker_to_querystring(url):
+    """For the given URL, check if it has a cachebreaking hash. If it does,
+    convert it to be a URLthat uses the hash as a querystring cachebreaker
+    instead.
+
+    eg
+        INPUT: 'https://example.com/static/foo.12313abc.jpg'
+        OUTPUT: 'https://example.com/static/foo.jpg?h=12313abc'
+
+    Why? So that images automatically embedded in social media posts (via OG
+    tags) don't end up as broken links with a new release that changes the hash
+    in the filename -- by moving the hash to a querystring, we get the benefit
+    of cachebreaking but with a much softer edge that won't result in links
+    404ing even if the cachebreaking hash has changed. (ie, querystring
+    cachebreakers are more forgiving.)
+
+    Note that ManifestFileStorage (used by django-whitenoise in this project) makes
+    hash-named versions of the files, but also _does_ keep the original around,
+    so we are safe to assume that if static/img/foo.213123.jpg exists, then
+    static/img/foo.jpg will, too
+    """
+
+    pattern = re.compile(r"(\.[a-f0-9]+)(\.\w+)$")
```

Comment:
I'm not the best regex-r so this is definitely more of a question:

Aren't the second set of parantheses superfluous, given that only the the first matching group is used?

Reply:
True - yes, that second group is (now) unnecessary -- originally i was going the replacement with via `pattern.sub` but it was just getting too complicated. Will remove the second grouping, but keep the search parameters - there are unit test to conifrm if i break things ;)